### PR TITLE
Make target for generating service clients

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Added a step to re-generate gRPC API clients under `/client`.

## Why?

Whenever proto files are re-compiled, the gRPC clients should also be re-generated.

Right now the developer has to manually invoke `go generate` (and know where to find them first).

## How did you test it?

Running `make proto` and `make update-proto` - and observing the clients being updated.

## Potential risks

There could be some subtle ordering assumption in `update-proto` that is violated now?

## Is hotfix candidate?

No

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,10 +15,10 @@ ci-build-misc: print-go-version ci-update-tools proto bins shell-check copyright
 clean: clean-bins clean-test-results
 
 # Recompile proto files.
-proto: clean-proto buf-lint api-linter protoc goimports-proto proto-mocks copyright-proto
+proto: clean-proto buf-lint api-linter protoc proto-clients goimports-proto proto-mocks copyright-proto
 
 # Update proto submodule from remote and recompile proto files.
-update-proto: clean-proto update-proto-submodule buf-lint api-linter protoc update-go-api goimports-proto proto-mocks copyright-proto gomodtidy
+update-proto: update-proto-submodule proto gomodtidy
```
Comment (stephanos) [inline Makefile:21]: drive-by fix: `update-proto` can re-use the `proto` target

File: Makefile [hunk 2]
```diff
@@ -207,6 +208,12 @@ proto-mocks: protoc
 		mockgen -copyright_file ../LICENSE -package $(call service_name,$(PROTO_GRPC_SERVICE))mock -source $(PROTO_GRPC_SERVICE) -destination $(call mock_file_name,$(PROTO_GRPC_SERVICE)) \
 	$(NEWLINE))
 
+proto-clients:
+	@printf $(COLOR) "Generate proto clients..."
+	$(foreach PROTO_GRPC_CLIENT,$(PROTO_GRPC_CLIENTS),\
+		@go generate $(PROTO_GRPC_CLIENT) \
+	$(NEWLINE))
```
Comment (dnr) [inline Makefile:215]: why not just
```suggestion
	@go generate ./client/...
```
?

Comment (stephanos) [reply Makefile:215]: Ha! Didn't know it could do that 👍

File: Makefile (hunk 1 above)
Comment (dnr) [reply Makefile:21]: In theory none of these are correct usage of make. Makefiles shouldn't be written assuming that dependencies will be "built" in any particular order. But I think as long as all of them are declared in `.PHONY` (you should add the ones that are missing) and no one uses `-j` then it'll continue to work.

File: Makefile [hunk 3]
```diff
@@ -207,6 +207,10 @@ proto-mocks: protoc
 		mockgen -copyright_file ../LICENSE -package $(call service_name,$(PROTO_GRPC_SERVICE))mock -source $(PROTO_GRPC_SERVICE) -destination $(call mock_file_name,$(PROTO_GRPC_SERVICE)) \
 	$(NEWLINE))
 
+proto-clients:
```
Comment (dnr) [inline Makefile:210]: Nitpicky, but I don't think this name makes sense. grpc already generates "proto clients", as in client stubs for services defined in proto[buf] files. The code in `client` adds metrics, retries (should be moved to interceptors), redirection, and most importantly routing, which are all application-level stuff, not "proto" stuff.

Maybe `service-clients`?

**Review by dnr — APPROVED** [review]
approved, but would prefer a different name

File: Makefile (hunk 3 above)
Comment (stephanos) [reply Makefile:210]: I was going back and forth on the name, I like `service-clients`.
